app/models/liabilitycalculation/viewmodels/TaxYearSummaryViewModel.scala

```diff
@@ -44,9 +43,35 @@ object TaxYearSummaryViewModel {
       deductions = calc.calculation.flatMap(c => c.taxCalculation.map(tc => tc.incomeTax.totalAllowancesDeductionsReliefs)).getOrElse(0.00),
       totalTaxableIncome = calc.calculation.flatMap(c => c.taxCalculation.map(_.incomeTax.totalTaxableIncome)).getOrElse(0),
       forecastIncome = calc.calculation.flatMap(c => c.endOfYearEstimate.flatMap(_.totalEstimatedIncome)),
-      forecastIncomeTaxAndNics = calc.calculation.flatMap(c => c.endOfYearEstimate.flatMap(_.incomeTaxNicAndCgtAmount))
+      forecastIncomeTaxAndNics = calc.calculation.flatMap(c => c.endOfYearEstimate.flatMap(_.incomeTaxNicAndCgtAmount)),
+      inYearCalculationRunDate = calc.metadata.calculationTimestamp.flatMap(timestamp => getInYearCalculationRunDate(timestamp, taxYear))
     )
   }
+
+  def isUnattendedCalc(calculationReason: Option[String]): Boolean = calculationReason match {
+    case Some("unattendedCalculation") => true
+    case _ => false
+  }
+
+  def getInYearCalculationRunDate(calculationTimestamp: String, taxYear: Int): Option[String] = {
+    val firstQuarterstart = LocalDate.parse(s"${taxYear - 1}-04-05")
+    val firstQuarterEnd = LocalDate.parse(s"${taxYear - 1}-07-05")
+    val SecondQuarterEnd = LocalDate.parse(s"${taxYear - 1}-10-05")
+    val thirdQuarterEnd = LocalDate.parse(s"${taxYear}-01-05")
+    val fourthQuarterEnd = LocalDate.parse(s"${taxYear}-04-05")
+
+    val instant = java.time.Instant.parse(calculationTimestamp)
+    val date = LocalDateTime.ofInstant(instant,ZoneId.of(ZoneOffset.UTC.getId)).toLocalDate
+
+    date match {
```

Comment:
I would guess this part of the logic need to be unit tests, I don't see unit tests for these; pls add